src/main/scala/io/github/pauljamescleary/petstore/repository/DoobiePetRepositoryInterpreter.scala

```diff
@@ -76,12 +76,26 @@ class DoobiePetRepositoryInterpreter[F[_]: Monad](val xa: Transactor[F])
       .list
       .transact(xa)
 
-  override def findByStatus(statuses: NonEmptyList[PetStatus]): F[List[Pet]] = {
-    val q = sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
+  def findByStatus(statuses: NonEmptyList[PetStatus]): F[List[Pet]] =
+    (sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
             FROM PET
-           WHERE """ ++ Fragments.in(fr"STATUS", statuses)
+           WHERE """ ++ Fragments.in(fr"STATUS", statuses))
+      .query[Pet]
+      .list
+      .transact(xa)
+
+  def findByTag(tags: NonEmptyList[String]): F[List[Pet]] = {
+    /* Handle dynamic construction of query based on multiple parameters */
 
-    q.query[Pet].list.transact(xa)
+    /* To piggyback off of comment of above reference about tags implementation, findByTag uses LIKE for partial matching
+    since tags is (currently) implemented as a comma-delimited string */
+    val tagLikeString: String = tags.toList mkString ("TAGS LIKE '%","%' OR TAGS LIKE '%","%'")
+    (sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
```

Comment:
do we need the parens?

Reply:
The parens are needed to join the SQL Interpolator with the fragment.

```
    (sql"""SELECT NAME, CATEGORY, BIO, STATUS, TAGS, PHOTO_URLS, ID
         FROM PET
         WHERE """ ++ Fragment.const(tagLikeString))
      .query[Pet]
      .list
      .transact(xa)
```